imagekit-sdk/src/main/java/io/imagekit/sdk/tasks/UrlGen.java

```diff
@@ -95,31 +111,24 @@ private static String buildPathUrl(String path, String urlEndpoint, Map<String,
         }
 
         URI baseUrl=URI.create(urlEndpoint);
-        URI newUri= null;
-        try {
+        URL newUri= null;
             String newPath=baseUrl.getPath();
-            if (tr.toString().equalsIgnoreCase("")){
-                newPath+=path.substring(1,path.length());
-            }
-            else {
-                newPath+=tr+path;
+            if (tr.toString().equalsIgnoreCase("")) {
+                newPath += path.substring(1);
+            } else {
+                newPath += tr + path;
             }
 
-            URI tmpUri = new URI(baseUrl.getScheme(),baseUrl.getUserInfo(),baseUrl.getHost(),baseUrl.getPort(),
-                    newPath,
-                    queryMaker.get(),null);
+            URL tmpUri = new URL(baseUrl.getScheme(), baseUrl.getHost(), baseUrl.getPort(),
+                    newPath + queryMaker.getAsQueryString(), null);
 
             if (signed){
-                sign(urlEndpoint, expireSeconds, privateKey, queryMaker, tmpUri);
+                sign(urlEndpoint, expireSeconds, privateKey, queryMaker, tmpUri.toURI());
             }
 
-            newUri = new URI(baseUrl.getScheme(),baseUrl.getUserInfo(),baseUrl.getHost(),baseUrl.getPort(),
-                    newPath,
-                    queryMaker.get(),null);
+            newUri = new URL(baseUrl.getScheme(), baseUrl.getHost(), baseUrl.getPort(),
```

Comment:
Isn't this exactly the same as above when you initially create the `tmpUri`?

Reply:
I'm not trying to refactor their copy-paste
👎